### PR TITLE
Add dispatcher event for remote fed shares

### PR DESCRIPTION
**Pull request by sharidas**
Adding dispatcher events for remote fed shares

Signed-off-by: Sujith H <sharidasan@owncloud.com>

## Description

Dispatcher event for remote fed shares.

## Motivation and Context

This change will help users to catch the events and log it.

## How Has This Been Tested?

- [x] Create a fed1 folder from a remote oC instance1
- [x] Share to another oC instance2.
- [x] When oC instance2 accepts/rejects the share the logs are available at oC2 instance.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by sharidas** [comment]
Backport for stable10 : https://github.com/owncloud/core/pull/29482

File: apps/files_sharing/lib/Controllers/ExternalSharesController.php [hunk 1]
```diff
@@ -56,6 +61,7 @@ public function __construct($appName,
 		parent::__construct($appName, $request);
 		$this->externalManager = $externalManager;
 		$this->clientService = $clientService;
+		$this->dispatcher = \OC::$server->getEventDispatcher();
```
Comment (PVince81) [inline apps/files_sharing/lib/Controllers/ExternalSharesController.php]: why not inject it directly ? you're in the perfect setup to do so as controllers are DI'd automagically

**Review by PVince81 — COMMENTED** [review]
Fix DI then this looks good

**Comment by sharidas** [comment]
Injected EventDispatcherInterface.

**Review by PVince81 — APPROVED** [review]
:+1:

**Pull request merged**